test/FileReadDataSource.test.js

```diff
@@ -0,0 +1,99 @@
+const expect = require('expect');
```

Comment:
Adjust those tests to fit any abstract datasource

Reply:
Created a responsive testing file with unic methods for all datasources. Needed some review